### PR TITLE
[IMP] Add product links smartbutton

### PR DESCRIPTION
**Pull request by ajaniszewska-dev**
aim of this PR is to add button with links to product.product form view.

### CODE REVIEW

**Review by simahawk — CHANGES_REQUESTED** [review]
LG but please add a test. There's one already on template.

File: product_variant_multi_link/models/product_product.py [hunk 1]
```diff
@@ -14,6 +15,47 @@ class ProductProduct(models.Model):
...
+
+    @api.depends("product_variant_link_ids")
+    def _compute_product_product_link_count(self):
+        link_model = self.env["product.template.link"]
+        # Set product_template_link_qty to 0 if user has no access on the model
+        try:
+            link_model.check_access_rights("read")
+        except AccessError:
+            for rec in self:
+                rec.product_product_link_count = 0
```
Comment (i-vyshnevska) [inline product_variant_multi_link/models/product_product.py]:
```suggestion
            self.updae({'product_product_link_count':0})
```

File: product_variant_multi_link/models/product_product.py [hunk 2]
```diff
@@ -14,6 +15,47 @@ class ProductProduct(models.Model):
...
+    product_product_link_count = fields.Integer(
+        string="Product Links Count", compute="_compute_product_product_link_count"
+    )
+
+    @api.depends("product_variant_link_ids")
+    def _compute_product_product_link_count(self):
+        link_model = self.env["product.template.link"]
+        # Set product_template_link_qty to 0 if user has no access on the model
+        try:
+            link_model.check_access_rights("read")
```
Comment (i-vyshnevska) [inline product_variant_multi_link/models/product_product.py:27]: or you can add a group right on xml view so for the rest of the users button will be invisible

Comment (simahawk) [reply product_variant_multi_link/models/product_product.py:27]: then if you use the same field in another view or piece of code you won't have the same result

**Review by i-vyshnevska — APPROVED** [review]

File: product_variant_multi_link/models/product_product.py [hunk 3]
```diff
@@ -14,6 +15,46 @@ class ProductProduct(models.Model):
         compute="_compute_product_link_ids",
     )
 
+    product_product_link_count = fields.Integer(
+        string="Product Links Count", compute="_compute_product_product_link_count"
```
Comment (sebalix) [inline product_variant_multi_link/models/product_product.py]: Runbot is complaining about this label:

    Two fields (product_template_link_count, product_product_link_count) of product.product() have the same label: Product Links Count.

Can you handle it? Maybe "Variant Links Count"? (I don't know the context)

Comment (ajaniszewska-dev) [reply product_variant_multi_link/models/product_product.py]: should be fine now.

**Review by simahawk — APPROVED** [review]

**Pull request merged**